docs/pages/extensibility/ui.md

```diff
@@ -39,8 +39,105 @@ Meshery leverages remote providers for identity management. These providers can
 
 The user avatar behavior, which changes based on the user's status, can be customized by extending the [User Component](https://github.com/meshery/meshery/blob/7de49ef4928f114080f923f2ad261f4433ca91d6/ui/components/User.js#L46).
 
+### Extensibility: Permissions (CASL)
 
-###  Build-Time UI Extensibility
+#### Design
+
+Meshery UI uses [CASL.js](https://casl.js.org/v6/en/) to implement it's permissions framework. CASL is an isomorphic authorization JavaScript library which restricts what resources a given client is allowed to access.
+Meshery's permissions framework has been designed to be very customizable and robust.
+
+- It uses keys as it's unit checks to see if a user has permission to perform action or not.
+- These keys are bundled together in keychains. Keychains are collections of similar permissions, which can themselves be grouped together and assigned to roles.
+- Roles map permissions to users. Roles contain any number of keychains, which contain any number of keys (permissions). Meshery uses roles to assign permissions to users.
+
+{% include alert.html type="info" title="Learn more" content="To learn more about how permissions work in Meshery Provider in terms of Roles, keychains and keys, head over to <a href='https://docs.layer5.io/cloud/security/'>Layer5 documentation</a>." %}
+
+#### Implementation
+
+- Once a user has logged in, the backend will send a response with the permissions that the user has, based on the role assigned to them. Those permissions will be used to check abilities on the frontend.
```

Comment:
backend?  Is that Meshery Server or the Provider or a combination?

frontend? You mean Meshery UI? 

When naming these architectural components, include hyperlinks to their concepts pages.